index.js

```diff
@@ -0,0 +1,185 @@
+'use strict'
+
+const { Readable, finished } = require('node:stream')
+const fs = require('node:fs')
+
+const ioDone = Symbol('ioDone')
+
+class TailFileStream extends Readable {
+  #watching = false
+  #waiting = false
+  #performingIO = false
+
+  #start
+  #end
+  #pos
+  #flags
+  #mode
+  #fd
+
+  #autoWatch
+  #persistent
+
+  constructor (path, options = {}) {
+    if (options.highWaterMark === undefined) {
+      options.highWaterMark = 64 * 1024
+    }
+
+    super(options)
+
+    this.path = path
+    this.#flags = options.flags ?? 'r'
+    this.#mode = options.mode ?? 0o666
+
+    this.#start = options.start ?? 0
+    this.#end = options.end ?? Infinity
+    this.#pos = this.#start
+    this.bytesRead = 0
+
+    this.#autoWatch = options.autoWatch ?? true
+    this.#persistent = options.persistent ?? true
+
+    this.#fd = null
+    this.watcher = null
+  }
+
+  get pending () {
+    return this.#fd === null
+  }
+
+  get watching () {
+    return this.#watching
+  }
+
+  get waiting () {
+    return this.#waiting
+  }
+
+  _construct (cb) {
+    if (this.#autoWatch) {
+      this.watch()
+    }
+
+    fs.open(this.path, this.#flags, this.#mode, (err, fd) => {
+      if (err) {
+        cb(err)
+      } else {
+        this.#fd = fd
+        cb()
+        this.emit('open', this.#fd)
+        this.emit('ready')
```

Comment:
```suggestion
        process.nextTick(() => {
          this.emit('open', this.#fd)
          this.emit('ready')
        })
```

See https://github.com/nodejs/node/issues/51993